services/galley/test/integration/API/Teams/LegalHold.hs

```diff
@@ -184,22 +186,48 @@ testCreateTwoLegalHoldDevices = do
 
 testApproveLegalHoldDevice :: TestM ()
 testApproveLegalHoldDevice = do
-    pure ()
+    (owner, tid) <- createTeam
+    member <- randomUser
+    addTeamMemberInternal tid $ newTeamMember member (rolePermissions RoleMember) Nothing
+
+    cannon <- view tsCannon
+
+    WS.bracketR cannon member $ \ws -> withDummyTestServiceForTeam owner tid $ do
+        -- not allowed to approve if team setting is disabled
+        -- TODO: remove the following 'ignore' once 'disabled' is the default
+        ignore $ approveLegalHoldDevice owner member tid !!! const 403 === statusCode
+
+        putEnabled tid LegalHoldEnabled
+        requestDevice owner member tid !!! const 204 === statusCode
+
+        -- Only the user themself can approve adding a LH device
+        approveLegalHoldDevice owner member tid !!! const 403 === statusCode
+        approveLegalHoldDevice member member tid !!! const 200 === statusCode
 
-    when False $ void $ approveLegalHoldDevice undefined undefined
-      -- just something silly to keep the name to suppress the `-Wunused-top-binds` noise and
-      -- to remind everybody it's already there.
+        cassState <- view tsCass
+        liftIO $ do
+            clients' <- Cql.runClient cassState $ Data.lookupClients [member]
+            print clients'
+            assertBool "Expect clientId to be saved on the user"
+              $ Clients.contains member someClientId clients'
```

Comment:
it's a bit weird that this is a global variable, but i think i commited that crime at least once in this module before...  :)

Reply:
The reason is because the clientId must match the fingerprint of the keys which we use (which are also a global variable) I added a comment on those saying they shouldn't be changed 😄